### PR TITLE
Add a helper function for parse out msg responses

### PR DESCRIPTION
**Pull request by phamminh0811**
## Description

closes: #4803 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: testing/utils.go [hunk 1]
```diff
@@ -33,3 +36,19 @@ func GenerateString(length uint) string {
...
+	}
+
+	for i, msg := range msgs {
+		if err := cdc.Unmarshal(txMsgData.MsgResponses[i].Value, msg); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```
Comment (chatton) [inline testing/utils.go:54]: thanks @phamminh0811! 

It looks like we have some very similar code in our [e2es](https://github.com/cosmos/ibc-go/blob/08d36eb0735caa29ddf981a872d0a55d218be3f6/e2e/testsuite/codec.go#L96-L120) at the moment, I wonder if we could maybe add the functionality here, and also refactor the E2E version to call into this, WDYT?

Comment (damiannolan) [reply testing/utils.go:54]: agree!

File: testing/utils.go [hunk 2]
```diff
@@ -33,3 +36,19 @@ func GenerateString(length uint) string {
 	}
 	return string(bytes)
 }
+
+// Parse out msg responses from a transaction result
```
Comment (damiannolan) [inline testing/utils.go:40]: nit: godoc should begin with func name

**Comment by hoangdv2429** [comment]
because of some unforeseen circumstances, @phamminh0811 no longer have access to notional fork to make the necessary changes so he will close this PR and I'll open another one.

File: testing/utils.go (hunk 1 above)
Comment (phamminh0811) [reply testing/utils.go:54]: got it, I'm gonna create a new pr and refactor that

**Pull request closed without merging**